support/scripts/mkukreloc.py

```diff
@@ -0,0 +1,325 @@
+#!/usr/bin/env python3
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2023, Unikraft GmbH and The Unikraft Authors.
+# Licensed under the BSD-3-Clause License (the "License").
+# You may not use this file except in compliance with the License.
+
+import argparse
+import subprocess
+import re
+
+# Generic regex for the type of an ELF section's entry
+TYPE_EXP = r'[a-zA-Z0-9_]+'
+
+# Generic regex for the name of an ELF section
+SEC_NAME_EXP = r'[a-zA-Z0-9_.\[\]]*'
+
+# Generic regex for a number in hexadecimal base
+HEXNUM_EXP = r'[a-f0-9]+'
+
+# Generic regex for a decimal number used as an index
+INDEX_EXP = r'[0-9]+'
+
+# Generic regex for a column separator
+SEP = r'\s*'
+
+# generic regex for a possible ELF symbol
+SYM_EXP = r'[a-zA-z0-9._]+'
+
+# Default values for out load address and the signature of the .ukreloc
+# section respectively (see reloc.h)
+BASE_ADDR = 0x0
+UKRELOC_SIGNATURE = 0xBADB0111
+
+# Rela Entries format from `readelf -r`:
+# Offset | Info | Type | Sym. Value | Sym. Name + Addend
+# However, since this is linked with -static-pie, all that we need can be
+# extracted from the .rela.dyn section(hopefully no .rel section since this
+# script does not account for those) as indicated by .dynamic. In our case
+# this section is expected to be filled with R_X86_64_RELATIVE or
+# R_AARCH64_RELATIVE (in the case of the latter, the E may not fit in the
+# command's output...).
+# Luckily we do not need to deal with .symtab since `readelf -r` nicely shows
+# the corresponding st_value.
+R_RELATIVE_TYPE_EXP = r'R_(?:X86_64|AARCH64)_RELATIV[E]{0,1}'
```

Comment:
That is done by this [check](https://github.com/unikraft/unikraft/pull/772/files#diff-1b6f1a9686178d25fb2b6e0f9ce8a803518660fed6a76c425c53a78df8e44acfR307). If the relocation is unexpected it will not be detected through the regex, leading to us finding less relocations than `.dynamic` reports.

Reply:
Yes I did not expect that this is done by the regex of course. Didn't notice the [check](https://github.com/unikraft/unikraft/pull/772/files#diff-1b6f1a9686178d25fb2b6e0f9ce8a803518660fed6a76c425c53a78df8e44acfR307), thanks 👍🏼